src/Request/ParamConverter/UpdateLineItemWebHookParamConverter.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Request\ParamConverter;
+
+use DateTimeImmutable;
+use OAT\SimpleRoster\Request\Validator\UpdateLineItemValidator;
+use OAT\SimpleRoster\WebHook\UpdateLineItemCollection;
+use OAT\SimpleRoster\WebHook\UpdateLineItemDto;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class UpdateLineItemWebHookParamConverter implements ParamConverterInterface
+{
+    /** @var UpdateLineItemValidator */
+    private $updateLineItemValidator;
+
+    public function __construct(UpdateLineItemValidator $updateLineItemValidator)
+    {
+        $this->updateLineItemValidator = $updateLineItemValidator;
+    }
+
+    public function apply(Request $request, ParamConverter $configuration): bool
+    {
+        $this->updateLineItemValidator->validate($request);
+
+        $responseBody = json_decode($request->getContent(), true);
+        $eventsRaw = $responseBody['events'];
+        $events = [];
+
+        foreach ($eventsRaw as $event) {
+            $events[] = new UpdateLineItemDto(
+                (string)$event['eventId'],
+                (string)$event['eventName'],
```

Comment:
About the logging, I do not know what you mean I checked other ParamConverters on this application and those aren't logging anything.
About the collection. It's not about the filter, it's about representing in a OOP a collection of objects. This class doesn't have the responsibility to filter any data, only to converter it. If we're filtering something here it's a violation of single responsibility.
About The DTO names, there is one important thing here, this is a webhook which it doesn't belong to our context and I wouldn't even consider it as REST endpoint concept. As you can see, this request was not designed thinking about our context (which'd be a PUT/PATCH of line items). It was designed to be more generic and let the client to decide what to do with the information. That's why I decided to put it on the DTO name because that's the purpose of a DTO object. I'm just encapsulating data to be represented between two systems.

Reply:
- If you want to log the request payload, then I'd go with a generic solution for all the endpoints. All of them do not log the payload. I do not see the point of only log this one. So my suggestion would be to have something like:
https://github.com/uniplaces/request-logger-bundle
- Collection: It's about avoiding primitive types by representing data and programming logic by using Object-oriented Program. There lots of benefits to using collections. Examples: logic encapsulated (you can't do it inside of primitive types), Code Responsibility (all of those logic operations are related to a specific object type, in this case, the DTOs), ....
- DTO: No this class name it's not about line item, I'm not mixing the domains here, this is exactly what I'm avoiding. I do not want to expose these events inside of our context, but I need to have these events data coming from tao